src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -90,6 +94,25 @@ public class FhirCosmosSearchService : SearchService
                 return new SearchResult(count, searchOptions.UnsupportedSearchParams);
             }
 
+            if (hasIncludeOrRevIncludeExpressions && (chainedExpressions.Count > 0))
+            {
+                List<Expression> chainedReferences = await PerformChainedSearch(searchOptions, chainedExpressions, cancellationToken);
+
+                if (chainedReferences.Any())
```

Comment:
TODO: Check if resulting `chainedReferences.Count` should eq `chainedExpressions.Count`. i.e. all chained queries should result in a filter